src/Tmds.Ssh/SshAgent.cs

```diff
@@ -23,60 +23,77 @@ sealed class SshAgent : IDisposable
         private const uint SSH_AGENT_RSA_SHA2_512 = 4;
         private const int MaxPacketSize = int.MaxValue; // We trust the SSH Agent.
 
-        private static EndPoint? _defaultEndPoint;
-
         public struct Identity
         {
             public string Comment { init; get; }
             public byte[] PublicKey { init; get; }
         }
 
-        public static EndPoint? DefaultEndPoint
+        public static string? DefaultEndPoint
         {
             get
             {
-                string? sshAuthSock = Environment.GetEnvironmentVariable("SSH_AUTH_SOCK");
-
-                if (OperatingSystem.IsWindows())
-                {
-                    return null;
-                }
-
-                if (string.IsNullOrEmpty(sshAuthSock))
-                {
-                    return null;
-                }
-
-                _defaultEndPoint ??= new UnixDomainSocketEndPoint(sshAuthSock);
-
-                return _defaultEndPoint;
+                return OperatingSystem.IsWindows()
+                    ? "openssh-ssh-agent"
+                    : Environment.GetEnvironmentVariable("SSH_AUTH_SOCK");
             }
         }
 
-        private readonly EndPoint _endPoint;
+        private readonly string _endPoint;
         private readonly SequencePool _sequencePool;
 
-        private SocketSshConnection? _agentConnection;
+        private StreamSshConnection? _agentConnection;
 
-        public SshAgent(EndPoint endPoint, SequencePool sequencePool)
+        public SshAgent(string endPoint, SequencePool sequencePool)
         {
             _endPoint = endPoint;
             _sequencePool = sequencePool;
         }
 
         public async ValueTask<bool> TryConnect(CancellationToken cancellationToken)
         {
+            NamedPipeClientStream? pipe = null;
             Socket? socket = null;
             try
             {
-                socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
-                await socket.ConnectAsync(_endPoint, cancellationToken).ConfigureAwait(false);
-                _agentConnection = new SocketSshConnection(NullLoggerFactory.Instance.CreateLogger<SshClient>(), _sequencePool, socket);
+                var logger = NullLoggerFactory.Instance.CreateLogger<SshClient>();
+
+                if (OperatingSystem.IsWindows())
+                {
+                    // There is no easy way to detect if a named pipe exists or
+                    // not. File.Exists is the fastest but will actually open a
+                    // connection and close it. Using this won't open the pipe
+                    // Directory.GetFiles("\\\\.\\pipe\\", endPoint).Length > 0
+                    // But it is slower.
+                    if (!File.Exists(@$"\\.\pipe\{_endPoint}"))
+                    {
+                        return false;
+                    }
+
+                    pipe = new NamedPipeClientStream(
+                        ".",
```

Comment:
This is needed to refer to localhost, the overload for asynchronous and anon impersonation require the server name to be set.